### DISCUSSION TITLE
Error 404 when submitting account creation form

### DISCUSSION THREAD

**Discussion by remiflament** [dani-garcia/vaultwarden#1974] [Q&A]
### Subject of the issue

When a user is invited to join an organization he receives an email link and fill the account creation form, it results in a 404 error.
The user can only see 

![2021-09-13_11-11](https://user-images.githubusercontent.com/8102922/133057704-0e1d8397-68c2-4a7c-9bf9-a83f5c15e159.png)

### Deployment environment

* vaultwarden version: official container vaultwarden/server:1.22.2

* Install method: Docker container pulled from hub.docker.com

* Clients used:  Web vault: tested with Chrome 93 & Firefox 92

* Reverse proxy and version:  No reverse proxy

* MySQL/MariaDB or PostgreSQL version:  PostgreSQL 12 managed in GCP

* Other relevant details: The Vaultwarden app in running on Cloud Run on GCP - Works perfectly since end of 2020. This bug is recent and **was initially found on 1.22.0**. I updated to check if its fix it, but not.

### Steps to reproduce

Here my modified parameters from the defaults values
USER_ATTACHMENT_LIMIT 0
ORG_ATTACHMENT_LIMIT 0
DATABASE_URL postgresql://user:pass@ip:port/bitwardenrs
SMTP_HOST smtp.confidential.com
SMTP_PORT XXX
SMTP_SSL true
SMTP_USERNAME user@confidential.com
SMTP_PASSWORD confidentialpasswordkey
SMTP_FROM bitwardenrs@confidential.com
DOMAIN https://bitwarden.confidential.com/
ADMIN_TOKEN TOKENADMINXXXX
SMTP_FROM_NAME Bitwarden
SIGNUPS_ALLOWED true
SIGNUPS_DOMAINS_WHITELIST confidential.com

I go to the organization page and invite people to join me.
![2021-09-13_11-30](https://user-images.githubusercontent.com/8102922/133059995-9f96f0e0-0e9c-4450-a824-b1f4ff454696.png)

They receive email link to join and click on it. Fill the form and get the 404 message in red, see above of this issue.
![2021-09-13_11-32](https://user-images.githubusercontent.com/8102922/133060469-93fe23e0-da87-4f62-b2e3-03a819b1f3c7.png)
![2021-09-13_11-33](https://user-images.githubusercontent.com/8102922/133060551-62248644-596b-4c8a-af1d-27d6130322df.png)

### Expected behaviour

The form should be sent and the account created.

### Actual behaviour

404 error when submitting the account creation form.

### Troubleshooting data

-----------------------------------------------
Thank you for this project!

**Comment by BlackDex** [comment]
Please provide the `Support String` you can generate on the /admin/diagnostics page.
Also, could you provide the log lines from vaultwarden when you try to execute this action?
And if possible increase the log_level to debug so that we would have a bit more information available.

Thanks.

**Comment by remiflament** [comment]
Copy paste of `Support String`

### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.22.2
* Web-vault version: v2.21.1
* Running within Docker: false
* Environment settings overridden: false
* Uses a reverse proxy: true
* IP Header check: false (X-Forwarded-For)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: PostgreSQL
* Database version: PostgreSQL 12.7 on x86_64-pc-linux-gnu, compiled by Debian clang version 10.0.1 , 64-bit
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** 


```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": true,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_ip_header_enabled": true,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_max_conns": 10,
  "database_url": "**********://****:***************@**.**.**.*:****/***********",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://***********.********.*****/",
  "domain_origin": "*****://***********.********.*****",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "enable_db_wal": true,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "invitation_org_name": "Vaultwarden",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "debug",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "org_attachment_limit": 0,
  "org_creation_users": "",
  "password_iterations": 100000,
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": true,
  "signups_domains_whitelist": "********.**",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_explicit_tls": false,
  "smtp_from": "***********@********.*****",
  "smtp_from_name": "Bitwarden",
  "smtp_host": "****-*****.**********.***",
  "smtp_password": "***",
  "smtp_port": 587,
  "smtp_ssl": true,
  "smtp_timeout": 15,
  "smtp_username": "******@********.**",
  "templates_folder": "data/templates",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_syslog": false,
  "user_attachment_limit": 0,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": false,
  "websocket_port": 3012,
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>

 set the debug_level to `debug`

I attach the few logs I get, they are all `info` level, zero debug log output with the action.

The action is: re-invite the member. Click on the new link to create account. Fill the form. Submit it.

re-invit
![re-invite](https://user-images.githubusercontent.com/8102922/133085494-e74069a0-26d5-4f89-8261-8d0a12b2546e.png)

![after-click](https://user-images.githubusercontent.com/8102922/133085523-72d2a041-39e5-462c-8f46-98e2ecbc0c76.png)

![after-click-create](https://user-images.githubusercontent.com/8102922/133085561-524d85ad-48ac-4211-9805-9ce4e6528576.png)

![after-submit](https://user-images.githubusercontent.com/8102922/133085612-198792b8-3300-4005-a858-d1372d5fe02f.png)

Thank you

**Comment by BlackDex** [comment]
I don't know what you are running which alters the URL's, but that `?_se=` part shouldn't be there.
So it looks like something is altering your URL's in a non compatible way.

As this does work for me without an issues.
I suggest to try this in an incognito/private browser (without any extensions if possible) and try it again.

**Reply by remiflament** [reply]
During the tests, I was already on incognito mode.
Maybe it's my SMTP provider how overrode the link. I will change it.

Thank you for your time.
